RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotFromDB.cc

```diff
@@ -0,0 +1,52 @@
+/**_________________________________________________________________
+   class:   OnlineBeamSpotFromDB.cc
+   package: RecoVertex/BeamSpotProducer
+   
+
+
+ author: Francisco Yumiceva, Fermilab (yumiceva@fnal.gov)
+
+
+________________________________________________________________**/
+
+// C++ standard
+#include <string>
+// CMS
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotFromDB.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/IOVSyncValue.h"
+#include "CondFormats/DataRecord/interface/BeamSpotTransientObjectsRcd.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+
+OnlineBeamSpotFromDB::OnlineBeamSpotFromDB(const edm::ParameterSet& iConfig) {}
+
+OnlineBeamSpotFromDB::~OnlineBeamSpotFromDB() {}
+
+void OnlineBeamSpotFromDB::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::ESHandle<BeamSpotObjects> beamGThandle;
+  edm::ESHandle<BeamSpotObjects> beamhandle;
+  iSetup.get<BeamSpotTransientObjectsRcd>().get(beamhandle);
+  const BeamSpotObjects* mybeamspot = beamhandle.product();
+  //iSetup.get<BeamSpotObjectsRcd>().get(beamGThandle);
+  //const BeamSpotObjects* myGTbeamspot = beamGThandle.product();
+
+  edm::LogInfo("Run numver: ") << iEvent.id().run();
+  edm::LogInfo("beamspot from HLT ") << *mybeamspot;
+  //edm::LogInfo("beamspot from GT ")<<*myGTbeamspot;
+}
+void OnlineBeamSpotFromDB::fillDescription(edm::ParameterSetDescription& desc) {}
```

Comment:
fillDescriptions is empty... need at least:
```
  edm::ParameterSetDescription dsc;
  desc.addWithDefaultLabel(dsc);
```
even if there are no parameters...

Reply:
Also wrong syntax, fillDescription argument is ```edm::ConfigurationDescriptions& descriptions```, not ```edm::ParameterSetDescription& desc```: see example:
```
void HLTBool::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
  edm::ParameterSetDescription desc;
  desc.add<bool>("result", false);
  descriptions.add("hltBool", desc);
}
```